src/main/scala/io/delta/execution/DeltaTableOperations.scala

```diff
@@ -68,4 +77,115 @@ trait DeltaTableOperations { self: DeltaTable =>
+   */
+  def update(set: java.util.Map[String, Column]): Unit = {
+    executeUpdate(set.asScala, None)
+  }
+
+  /**
+   * Update data from the table on the rows that match the given `condition`,
+   * which performs the rules defined by `set`.
+   * @param condition update operations will be performed if rows hold true for this condition.
+   * @param set specifies the rules for how to update a row.
+   */
+  def update(condition: Column, set: Map[String, Column]): Unit = {
+    executeUpdate(set, Some(condition))
+  }
+
+  /**
+   * Update data from the table on the rows that match the given `condition`,
+   * which performs the rules defined by `set`.
+   * Java friendly API.
+   * @param condition update operations will be performed if rows hold true for this condition.
+   * @param set specifies the rules for how to update a row.
+   */
+  def update(condition: Column, set: java.util.Map[String, Column]): Unit = {
+    executeUpdate(set.asScala, Some(condition))
+  }
+
+  /**
+   * Update data from the table, which performs the rules defined by `set`.
+   * @param set specifies the rules for how to update a row.
+   */
+  def updateExpr(set: Map[String, String]): Unit = {
+    executeUpdate(toStrColumnMap(set), None)
+  }
+
+  /**
+   * Update data from the table, which performs the rules defined by `set`.
+   * Java friendly API.
+   * @param set specifies the rules for how to update a row.
+   */
+  def updateExpr(set: java.util.Map[String, String]): Unit = {
+    executeUpdate(toStrColumnMap(set.asScala), None)
+  }
+
+  /**
+   * Update data from the table on the rows that match the given `condition`,
+   * which performs the rules defined by `set`.
+   * @param condition update operations will be performed if rows hold true for this condition.
+   * @param set specifies the rules for how to update a row.
+   */
+  def updateExpr(condition: String, set: Map[String, String]): Unit = {
+    executeUpdate(toStrColumnMap(set), Some(functions.expr(condition)))
+  }
+
+  /**
+   * Update data from the table on the rows that match the given `condition`,
+   * which performs the rules defined by `set`.
+   * Java friendly API.
+   * @param condition update operations will be performed if rows hold true for this condition.
+   * @param set specifies the rules for how to update a row.
+   */
+  def updateExpr(condition: String, set: java.util.Map[String, String]): Unit = {
+    executeUpdate(toStrColumnMap(set.asScala), Some(functions.expr(condition)))
+  }
+
+  protected def toStrColumnMap(map: Map[String, String]): Map[String, Column] = {
+    map.toSeq.map { case (k, v) => k -> functions.expr(v) }.toMap
+  }
+
+  protected def makeUpdateTable(
+      sparkSession: SparkSession,
+      target: DeltaTable,
+      onCondition: Option[Column],
+      setColumns: Seq[(String, Column)]): UpdateTable = {
+    val updateColumns = setColumns.map { x => UnresolvedAttribute.quotedString(x._1) }
```

Comment:
Why this but not `x => UnresolvedAttribute(x._1)`? what's the difference?

It would be good to have some tests for different variations of column references as this doesn't go through the normal parsing, e.g.,
```
colName (unquoted)
`colName` (quoted)
tableName.colName
`tableName`.`colName`
...
```

Reply:
please remember this in a follow up PR with nested field support.

Reply:
added.

Reply:
done.